### ISSUE TITLE
[7.3.1] Breaks Reflector

### ISSUE THREAD

**Issue by johannesschobel** [nestjs/nest#5024]
## Potential Commit/PR that introduced the regression

Somewhere between 7.2.0 and 7.3.0

## Describe the regression

The `Reflector` class does not work as before.

I have an `@Unprotected()` guard, that allows for calling a specific route - without a valid JWT. Respective Guard looks like this:

```ts
@Injectable()
export class GqlUnprotectedGuard implements CanActivate {
  constructor(private readonly reflector: Reflector) {}

  canActivate(context: ExecutionContext): boolean {
    const unprotected = this.reflector.getAllAndOverride<boolean>(
      DECORATOR_METADATA.API_UNPROTECTED,
      [context.getHandler(), context.getClass()],
    );

    if (!unprotected) {
      return false;
    }

    return unprotected;
  }
}
```

A test for this guards looks as follows:

```ts
class ReflectorMock {
  handlerUnprotected: boolean | undefined;
  classUnprotected: boolean | undefined;

  getAllAndOverride = jest.fn((metaDataKey: string, targets: any[]) => {
    return this.handlerUnprotected || this.classUnprotected;
  });
}

describe('UnprotectedGuard', () => {
  let gqlUnprotectedGuard: GqlUnprotectedGuard;
  let context: ExecutionContext;
  let reflector: ReflectorMock;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [
        GqlUnprotectedGuard,
        Reflector,
        ExecutionContextHost,
        {
          provide: Reflector,
          useClass: ReflectorMock,
        },
      ],
    }).compile();

    gqlUnprotectedGuard = module.get<GqlUnprotectedGuard>(GqlUnprotectedGuard);
    gqlUnprotectedGuard.getRequest = jest.fn(() => {});
    context = module.get<ExecutionContext>(ExecutionContextHost);
    reflector = module.get<any>(Reflector);
  });

  it('should disallow access if no decorator is assigned', async () => {
    const accessAllowed = gqlUnprotectedGuard.canActivate(context);
    expect(accessAllowed).toBe(false);
  });

  it('should allow access if a decorator is assigned to this method', async () => {
    reflector.handlerUnprotected = true;
    reflector.classUnprotected = undefined;

    const accessAllowed = gqlUnprotectedGuard.canActivate(context);
    expect(accessAllowed).toBe(true);
  });

  // ...
});
```

## Expected behavior/code

Respective code works in `Nest 7.2.0` but not in `Nest 7.3.0 / 7.3.1`.

## Environment
<pre><code>
Nest version: 7.2.0 -> 7.3.0

For Tooling issues:
- Node version: 12.18.2
- Platform: Linux, Ubuntu
</code></pre>

**Comment by hantsy** [comment]
Encountered the same problem. https://github.com/hantsy/nestjs-sample/issues/53

**Comment by kamilmysliwiec** [comment]
This should be fixed in 7.3.2. Please, let me know if you still face this issue.

**Comment by johannesschobel** [comment]
Dear @kamilmysliwiec ,

thank you very much for the support. The issue is resolved!
